### PR TITLE
Quantiles aggregation

### PR DESCRIPTION
**Pull request by g0djan**

### CODE REVIEW

**Comment by g0djan** [comment]
image

**Comment by reshke** [comment]
make apply_fmt

**Review by reshke — APPROVED** [review]

File: sources/console.c [hunk 1]
```diff
@@ -496,14 +496,53 @@ od_console_show_version(machine_msg_t *stream)
 	return rc;
 }
 
+static inline int
```
Comment (reshke) [inline sources/console.c:499]: static inline od_retcode_t

File: sources/console.c [hunk 2]
```diff
@@ -496,14 +496,53 @@ od_console_show_version(machine_msg_t *stream)
...
+		if (isnan(query_quantile)) {
+			query_quantile = 0;
+		}
+		if (isnan(transaction_quantile)) {
+			transaction_quantile = 0;
+		}
+		data_len =
+		  od_snprintf(data, sizeof(data), "%" PRIu64, (uint64_t)query_quantile);
+		rc = kiwi_be_write_data_row_add(stream, offset, data, data_len);
+		if (rc == -1)
```
Comment (reshke) [inline sources/console.c:524]: rc != OK_RESPONCE

File: sources/console.c [hunk 3]
```diff
@@ -883,12 +910,55 @@ od_console_show_pools(od_client_t *client, machine_msg_t *stream, bool extended)
...
+			goto error;
+		}
+	}
+	td_safe_free(transactions_hgram);
+	td_safe_free(queries_hgram);
 	return kiwi_be_write_complete(stream, "SHOW", 5);
+error:
+	td_safe_free(transactions_hgram);
+	td_safe_free(queries_hgram);
+	return -1;
```
Comment (reshke) [inline sources/console.c:961]: return NOT_OK_RESPONCE

File: third_party/kiwi/kiwi/be_write.h [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #ifndef KIWI_BE_WRITE_H
 #define KIWI_BE_WRITE_H
 
+#define EMPTY_MSG_LEN -1
```
Comment (reshke) [inline third_party/kiwi/kiwi/be_write.h:4]: :+1: :1st_place_medal: :100: :cool: :sunglasses: 
but, maybe 0?

Comment (g0djan) [reply third_party/kiwi/kiwi/be_write.h:4]: Unfortunately, nope. It is used in kiwi_be_write_data_row_add and logic has already based on using -1.

Comment (g0djan) [reply third_party/kiwi/kiwi/be_write.h:4]: Renamed EMPTY_MSG_LEN -> NULL_MSG_LEN
¯\_(ツ)_/¯

**Pull request merged**